magellan-library/src/main/java/com/wealthfront/magellan/navigation/LinearNavigator.kt

```diff
@@ -55,13 +49,10 @@ class LinearNavigator(
   }
 
   override fun onDestroy(context: Context) {
-    currentNavigable?.let {
-      lifecycleStateMachine.transitionBetweenLifecycleStates(it, Created(context), Destroyed)
+    backStack.navigables().forEach {
+      removeFromLifecycle(it, detachedState = Destroyed)
     }
-    lifecycleStateMachine.transitionBetweenLifecycleStates(
-      backStack.navigables(),
-      Created(context),
-      Destroyed)
+    backStack.clear()
```

Comment:
remove the navigables from the lifecycle and clear the backstack whenever the application onDestroy is called.